unreal_asset/src/reader/asset_trait.rs

```diff
@@ -13,7 +13,7 @@ pub trait AssetTrait {
     fn get_custom_version<T>(&self) -> CustomVersion
     where
         T: CustomVersionTrait + Into<i32>;
-    fn position(&self) -> u64;
+    fn position(&mut self) -> u64;
```

Comment:
Fetching position should never mutate state

Reply:
it doesn't change position at all though - it just calls seek(SeekFrom::Current(0))

Reply:
it still shouldn't force you to pass a mutable reference, so another solution must be found

Reply:
but it's only used internally so why should it matter?

Reply:
even internal apis should be correct and shouldn't require mutability on methods that do not mutate state